Windows removes wmic

**Describe the bug**
Windows 11 22483 removed wmic, so this software does not work anymore. 

**Steps to reproduce**
1. Update to the latest windows 11 version

2. Try using MWD or wmic in the cmd

**Log output**
```
Something went wrong while trying to authenticate your Minecraft account.

Authentication details are retrieved from the Minecraft game directly. Make sure that the game is running.
If you have a username set in the authentication tab, ensure that it matches the one you are trying to connect with.
If you are in an environment where this does not work, please set the correct username and token in the authentication tab.
See https://github.com/mircokroon/minecraft-world-downloader/wiki/Authentication for more details.

javax.security.sasl.AuthenticationException: Cannot get valid authentication details. [Caused by javax.security.sasl.AuthenticationException: Cannot find authentication details.]
	at proxy.auth.ClientAuthenticator.makeRequest(ClientAuthenticator.java:40)
	at proxy.EncryptionManager.lambda$sendReplacementEncryptionConfirmation$6(EncryptionManager.java:225)
	at proxy.EncryptionManager.disconnectOnError(EncryptionManager.java:113)
	at proxy.EncryptionManager.sendReplacementEncryptionConfirmation(EncryptionManager.java:225)
	at proxy.EncryptionManager.lambda$setClientEncryptionConfirmation$5(EncryptionManager.java:213)
	at proxy.EncryptionManager.attempt(EncryptionManager.java:100)
	at proxy.EncryptionManager.setClientEncryptionConfirmation(EncryptionManager.java:202)
	at packets.handler.ServerBoundLoginPacketHandler.lambda$new$1(ServerBoundLoginPacketHandler.java:30)
	at packets.handler.PacketHandler.handle(PacketHandler.java:57)
	at packets.DataReader.readPackets(DataReader.java:163)
	at packets.DataReader.pushData(DataReader.java:118)
	at proxy.ProxyServer.lambda$run$4(ProxyServer.java:86)
	at util.ExceptionHandling.attempt(ExceptionHandling.java:12)
	at proxy.ProxyServer.lambda$run$6(ProxyServer.java:83)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: javax.security.sasl.AuthenticationException: Cannot find authentication details.
	at proxy.auth.AuthDetailsManager.getAuthDetails(AuthDetailsManager.java:47)
	at proxy.auth.ClientAuthenticator.makeRequest(ClientAuthenticator.java:37)
	... 14 more
java.lang.IllegalStateException: Packet parsing may have been incorrect! Expected length: 261. Used bytes: 0
	at packets.DataReader.readPackets(DataReader.java:172)
	at packets.DataReader.pushData(DataReader.java:118)
	at proxy.ProxyServer.lambda$run$4(ProxyServer.java:86)
	at util.ExceptionHandling.attempt(ExceptionHandling.java:12)
	at proxy.ProxyServer.lambda$run$6(ProxyServer.java:83)
	at java.base/java.lang.Thread.run(Thread.java:831)
...

```
**Screenshots**
I tried to manually install wmic, but wmic is dead for the newest 11 version now

This is german, but you can see that this does not work
![image](https://user-images.githubusercontent.com/67484093/138927395-319960a7-9c1c-41ee-bd5a-e7c377b5bfc5.png)

**Version**
 — Game: Minecraft 1.17.1
 — Downloader: 1.14.8
 — Launcher (if login issues): Not sure where I can find it, but should be the latest

--- a/src/main/java/proxy/auth/AuthDetailsFromProcess.java
+++ b/src/main/java/proxy/auth/AuthDetailsFromProcess.java
@@ -128,29 +128,21 @@ public class AuthDetailsFromProcess {
     }
 
     /**
-     * Retrieve launch parameters from a given PID on Windows.
-     *
-     * Source: https://stackoverflow.com/a/54790608
+     * Retrieve launch parameters from a given PID on Windows. Uses PowerShell since WMIC is removed in windows 11.
      */
     private String getLaunchParametersWindows(long pid) {
+        String query = "SELECT CommandLine FROM Win32_Process WHERE ProcessID = \"" + pid + "\"";
+        String formatter = "| Select-Object -Property CommandLine | Out-String -width 9999";
+        String cmd = "\"Get-WmiObject -Query \\\"" + query + "\\\" " + formatter + "\"";
         try {
             Process process = new ProcessBuilder(
-                    "wmic", "process", "where", "ProcessID=" + pid, "get", "commandline", "/format:list"
+                    "Powershell.exe", "-Command", cmd
             ).redirectErrorStream(true).start();
 
-            try (InputStreamReader inputStreamReader = new InputStreamReader(process.getInputStream());
-                 BufferedReader reader = new BufferedReader(inputStreamReader)) {
-                while (true) {
-                    String line = reader.readLine();
-                    if (line == null) {
-                        return "";
-                    }
-                    if (!line.startsWith("CommandLine=")) {
-                        continue;
-                    }
-                    return line.substring("CommandLine=".length());
-                }
-            }
+            String res = new BufferedReader(new InputStreamReader(process.getInputStream()))
+                    .lines().filter(line -> line.contains("accessToken")).collect(Collectors.joining());
+
+            return res;
         } catch (IOException e) {
             e.printStackTrace();
             return "";

